src/Drivers/Zarinpal/Strategies/Normal.php

```diff
@@ -161,13 +161,16 @@ public function verify() : ReceiptInterface
 
         $result = json_decode($response->getBody()->getContents(), true);
 
-        if (empty($result['data']) || ! isset($result['data']['ref_id']) || $result['data']['code'] != 100) {
-            $message = $result['errors']['message'] ?? "";
+        if (empty($result['data']) || ! isset($result['data']['ref_id']) || ($result['data']['code'] != 100 && $result['data']['code'] != 101)) {
+            $message = $result['errors']['message'];
             $code = $result['errors']['code'];
-
             throw new InvalidPaymentException($message, $code);
         }
-
+        if ($result['data']['code'] == 101) {
```

Comment:
move this conditional statement to like 164 and remove `&& $result['data']['code'] != 101)`